astropy/modeling/core.py

```diff
@@ -2744,11 +2745,13 @@ def _initialize_slices(self):
             param_metrics[name]["shape"] = param_shape
             param_metrics[name]["size"] = param_size
             total_size += param_size
-        self._parameters = np.empty(total_size, dtype=np.float64)
+        if self._parameters is None or self._parameters.size != total_size:
```

Comment:
I would add a comment here to explain the reason this is conditional (perfs)

Reply:
Done